plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -1688,6 +1696,38 @@ public function add_price_range_filter_posts_clauses( $clauses, $query ) {
 		return $clauses;
 	}
 
+	/**
+	 * Add the `posts_clauses` filter to the main query.
+	 *
+	 * @param array    $clauses The query clauses.
+	 * @param WP_Query $query   The WP_Query instance.
+	 */
+	public function add_price_sorting_posts_clauses( $clauses, $query ) {
+		$query_vars                  = $query->query_vars;
+		$is_product_collection_block = $query_vars['isProductCollection'] ?? false;
+
+		if ( ! $is_product_collection_block ) {
+			return $clauses;
+		}
+
+		$orderby = $query_vars['orderby'] ?? null;
+
+		if ( 'price' !== $orderby ) {
+			return $clauses;
+		}
+
+		$clauses['join'] = $this->append_product_sorting_table_join( $clauses['join'] );
+		$order           = $query_vars['order'] ?? 'desc';
+
+		if ( 'asc' === strtolower( $order ) ) {
+			$clauses['orderby'] = ' wc_product_meta_lookup.min_price ASC, wc_product_meta_lookup.product_id ASC ';
+		} else {
+			$clauses['orderby'] = ' wc_product_meta_lookup.max_price DESC, wc_product_meta_lookup.product_id DESC ';
+		}
+
+		return $clauses;
+	}
+
```

Comment:
I am leaving some minor nitpicks below. Feel free to pick part of it or ignore it:

```suggestion
	/**
	 * Add the `posts_clauses` filter to add price-based sorting
	 *
	 * @param array    $clauses The list of clauses for the query.
	 * @param WP_Query $query   The WP_Query instance.
	 * @return array   Modified list of clauses.
	 */
	public function add_price_sorting_posts_clauses( $clauses, $query ) {
		$query_vars                  = $query->query_vars;
		$is_product_collection_block = $query_vars['isProductCollection'] ?? false;

		if ( ! $is_product_collection_block ) {
			return $clauses;
		}

		$orderby = $query_vars['orderby'] ?? null;
		if ( 'price' !== $orderby ) {
			return $clauses;
		}

		$clauses['join']    = $this->append_product_sorting_table_join( $clauses['join'] );
		$is_ascending_order = 'asc' === strtolower( $query_vars['order'] ?? 'desc' );

		$clauses['orderby'] = $is_ascending_order ?
			'wc_product_meta_lookup.min_price ASC, wc_product_meta_lookup.product_id ASC' :
			'wc_product_meta_lookup.max_price DESC, wc_product_meta_lookup.product_id DESC';

		return $clauses;
	}

```